volatility3/framework/plugins/linux/proc.py

```diff
@@ -35,16 +40,138 @@ def get_requirements(cls):
+            A list of vmas based on the task and filtered based on the filter function
+        """
+        if task.mm:
+            for vma in task.mm.get_mmap_iter():
+                if not filter_func(vma):
+                    yield vma
+
+    @classmethod
+    def vma_dump(
+        cls,
+        context: interfaces.context.ContextInterface,
+        task: interfaces.objects.ObjectInterface,
+        vma: interfaces.objects.ObjectInterface,
+        open_method: Type[interfaces.plugins.FileHandlerInterface],
+        maxsize: int = MAXSIZE_DEFAULT,
+    ) -> Optional[interfaces.plugins.FileHandlerInterface]:
+        """Extracts the complete data for VMA as a FileInterface.
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            task: an task_struct instance
+            vma: The suspected VMA to extract (ObjectInterface)
+            open_method: class to provide context manager for opening the file
+            maxsize: Max size of VMA section (default MAXSIZE_DEFAULT)
+
+        Returns:
+            An open FileInterface object containing the complete data for the task or None in the case of failure
+        """
+        try:
+            vm_start = vma.vm_start
+            vm_end = vma.vm_end
+        except AttributeError:
+            vollog.debug("Unable to find the vm_start and vm_end")
+            return None
+        
+        vm_size = vm_end - vm_start
+        if 0 < maxsize < vm_size:
+            vollog.debug(
+                f"Skip virtual memory dump {vm_start:#x}-{vm_end:#x} due to maxsize limit"
+            )
+            return None
+
+        pid = "Unknown"
+        try:
+            pid = task.tgid
+            proc_layer_name = task.add_process_layer()
+        except exceptions.InvalidAddressException as excp:
+            vollog.debug(
+                "Process {}: invalid address {} in layer {}".format(
+                    pid, excp.invalid_address, excp.layer_name
+                )
+            )
+            return None
+
+        proc_layer = context.layers[proc_layer_name]
+        file_name = f"pid.{pid}.vma.{vm_start:#x}-{vm_end:#x}.dmp"
+        try:
+            file_handle = open_method(file_name)
+            chunk_size = 1024 * 1024 * 10
+            offset = vm_start
+            while offset < vm_start + vm_size:
+                to_read = min(chunk_size, vm_start + vm_size - offset)
+                data = proc_layer.read(offset, to_read, pad=True)
+                if not data:
+                    break
+                file_handle.write(data)
+                offset += to_read
+
+        except Exception as excp:
+            vollog.debug(f"Unable to dump virtual memory {file_name}: {excp}")
+            return None
+
+        return file_handle
+
     def _generator(self, tasks):
+        # build filter for addresses if required
+        address_list = self.config.get("address", [])
+        if address_list == []:
```

Comment:
This is a little ugly, and I'd prefer the default were to be `None` and then we can definitively checked for `None` (or even `not address_list`) just in case the flag was provided but no items given somehow.

Reply:
I think that's because when a list option is provided it always returns an empty list even if the option isn't provided. Given your comment maybe that's not intended? 

e.g. this still returns an empty list
```
print(self.config["address"], type(self.config["address"]))
address_list = self.config.get("address", None)
print(address_list, type(address_list))
```

Here are the different case:
```
python vol.py -f linux-sample-1.dmp  linux.proc
Volatility 3 Framework 2.4.1
Progress:  100.00               Stacking attempts finished
PID     Process Start   End     Flags   PgOff   Major   Minor   Inode   File Path       File output
[] <class 'list'>
[] <class 'list'>
<snip>

python vol.py -f linux-sample-1.dmp  linux.proc --address
Volatility 3 Framework 2.4.1
Progress:  100.00               Stacking attempts finished
PID     Process Start   End     Flags   PgOff   Major   Minor   Inode   File Path       File output
[] <class 'list'>
[] <class 'list'>
<snip>

python vol.py -f linux-sample-1.dmp  linux.proc --address 1, 2, 3
Volatility 3 Framework 2.4.1
Progress:  100.00               Stacking attempts finished
PID     Process Start   End     Flags   PgOff   Major   Minor   Inode   File Path       File output
[1, 2, 3] <class 'list'>
[1, 2, 3] <class 'list'>
<snip>
```

I tried looking into why that happens, but got lost in requirements and configuration interfaces.

I will use `not address_list` for the moment.

Reply:
Sorry, that's what I meant, `not address_list` catches both an empty list and `None`, seems better than specifically checking equality against the empty list, thanks.  5:)